### PR TITLE
Routing updated

### PR DESCRIPTION
**Pull request by macdonst**
What do you think about this additional routing information? I'm a little concerned a changed a few routes. If we merge in this PR should we do redirects?

Signed-off-by: macdonst <simon.macdonald@gmail.com>

### CODE REVIEW

File: app/docs/md/learn/concepts/routing/dynamic-routes.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  return {
+    json: {
+     user
+    }
+  }
+}
+```
+</doc-code>
+
+Routes like `/users/axol` or `/users/admin` will be matched by `app/api/users/$id.mjs`. The `user` JSON object will be returned to the caller if the request's headers include `Accept: application/json` otherwise the route will call `app/pages/users/$id.mjs` if the file exists.
```
Comment (colepeters) [inline app/docs/md/learn/concepts/routing/dynamic-routes.md:26]: Noob question: if the route ends up calling `app/pages/users/$id.mjs`, will the `id` parameter be exposed to SFCs on that page in some way automatically? Or would I still need to author an API route to manage that?

**Comment by colepeters** [comment]
FWIW, these docs read great to me! Will leave it for the requested reviewers to approve or otherwise.

File: app/docs/md/learn/concepts/routing/dynamic-routes.md (hunk 1 above)
Comment (ryanbethel) [reply app/docs/md/learn/concepts/routing/dynamic-routes.md:26]: You would need an API to put `req.params` into the store.

**Review by ryanbethel — COMMENTED** [review]
Looks good. I think we just need to add a note somewhere showing the relative priority btw dynamic routes and catchall routes.

File: app/docs/md/learn/concepts/routing/dynamic-routes.md (hunk 1 above)
Comment (colepeters) [reply app/docs/md/learn/concepts/routing/dynamic-routes.md:26]: Makes sense, thanks!

**Comment by macdonst** [comment]
> Looks good. I think we just need to add a note somewhere showing the relative priority btw dynamic routes and catchall routes.

Good catch, I tried to make it a bit more clear here before boarding.

**Review by tbeseda — APPROVED** [review]
image

**Pull request merged**